Makefile

```diff
@@ -1,110 +1,93 @@
+COVERAGE_REPORT_FILE=coverage.txt
+COVERAGE_FLAGS=-coverprofile=$(COVERAGE_REPORT_FILE) -covermode=atomic
 
-all: test docs install
-.PHONY: all
+help: ## display help for this makefile
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+.PHONY: help
 
 setup: ## setup development dependencies
-	curl -sfL https://raw.githubusercontent.com/chanzuckerberg/bff/main/download.sh | sh
-	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh
-	curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh
+	@which ./bin/golangci-lint || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ./bin v1.53.3
+	@which ./bin/reviewdog || curl -sSfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b ./bin v0.14.2
 .PHONY: setup
 
-sweep:
-	@echo "WARNING: This will destroy infrastructure. Use only in development accounts."
-	SNOWFLAKE_ENABLE_SWEEP=1 go test -timeout 300s -run ^TestSweepAll ./pkg/sdk -v
+cleanup: ## cleanup development dependencies
+	rm -rf bin/*
+.PHONY: cleanup
 
-lint:  ## run the fast go linters
-	./bin/reviewdog -conf .reviewdog.yml  -diff "git diff main"
-.PHONY: lint
+sweep: ## destroy architecture; USE ONLY FOR DEVELOPMENT ACCOUNTS
+	@echo "WARNING: This will destroy infrastructure. Use only in development accounts."
+	@read -p "Are you sure? [y/n]" -n 1 REPLY; echo; \
+		if [[ $$REPLY =~ ^[yY]$$ ]]; then \
+		  SNOWFLAKE_ENABLE_SWEEP=1 go test -timeout 300s -run ^TestSweepAll ./pkg/sdk -v; \
+		  else echo "Aborting..."; \
+		fi;
+.PHONY: sweep
 
 lint-ci: ## run the fast go linters
 	./bin/reviewdog -conf .reviewdog.yml -reporter=github-pr-review -tee -fail-on-error=true
 .PHONY: lint-ci
 
-lint-all:  ## run the fast go linters
-	./bin/reviewdog -conf .reviewdog.yml  -filter-mode nofilter
-.PHONY: lint-all
-
-lint-missing-acceptance-tests:
-	@for r in `ls pkg/resources/ | grep -v list_expansion | grep -v privileges | grep -v grant_helpers | grep -v test | xargs -I{} basename {} .go`; do \
-		if [ ! -f pkg/resources/"$$r"_acceptance_test.go ]; then \
-			echo $$r; \
-		fi; \
-	done
-.PHONY: lint-missing-acceptance-tests
-
-build: ## build the binary
-	go build ${LDFLAGS} -o $(BASE_BINARY_NAME) .
-.PHONY: build
-
-coverage: ## run the go coverage tool, reading file coverage.out
-	go tool cover -html=coverage.txt
-.PHONY: coverage
-
-test:  ## run the tests (except sdk tests)
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/resources/...
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/provider/...
-	CGO_ENABLED=1 $(go_test) -race -coverprofile=coverage.txt -covermode=atomic $(TESTARGS) ./pkg/snowflake/...
-
+test: ## run the tests (except sdk tests)
+	CGO_ENABLED=1 go test -race $(COVERAGE_FLAGS) ./pkg/resources/...
```

Comment:
these are only unit tests which are now run by Unit tests giuthub action (defined in file `unit.yml`). TBH I would like to remove it also, and remove it from unit.yml, but I was not sure if someone is using it independently (I wrote my concerns in the doc that I have shared internally).

Reply:
I can remove it also, why not :)

Reply:
The name "unit test" is confusing, because its not including any unit tests from SDK. There may be some minimal "unit tests" that test specific helper functions / validation functions for the provider,

perhaps we should create two different workflows to test the SDK and the Terraform Provider. That would make it easier to split later. And then these makefile commands could be more specific such as `make unit-test provider` or something to that effect.

Reply:
Removed.